src/renderer/material/geometry_instance_material.rs

```diff
@@ -0,0 +1,50 @@
+use crate::core::*;
+use crate::renderer::*;
+
+///
+/// Used for rendering a unique color for each instance of an object with this material,
+/// as to allow uniquely determining which pixels belong to which object instance.
+///
+#[derive(Default, Clone)]
+pub struct GeometryInstanceMaterial {
+    /// Geometry ID
+    pub id: usize,
+    /// Render states.
+    pub render_states: RenderStates,
+}
+
+impl FromCpuMaterial for GeometryInstanceMaterial {
+    fn from_cpu_material(_context: &Context, _cpu_material: &CpuMaterial) -> Self {
+        Self::default()
+    }
+}
+
+impl Material for GeometryInstanceMaterial {
+    fn id(&self) -> u16 {
+        EffectMaterialId::GeometryInstanceMaterial.0
+    }
+
+    fn fragment_shader_source(&self, _lights: &[&dyn Light]) -> String {
+        include_str!("shaders/geometry_instance_material.frag").to_string()
+    }
+
+    fn fragment_attributes(&self) -> FragmentAttributes {
+        FragmentAttributes {
+            instance_id: true,
+            ..FragmentAttributes::NONE
```

Comment:
I guess you also need the position?

Reply:
Yep, forgot to add it back to the required attributes when I added it back to the shader.